pymc_extras/model/marginal/distributions.py

```diff
@@ -43,6 +55,74 @@ def support_axes(self) -> tuple[tuple[int]]:
             )
         return tuple(support_axes_vars)
 
+    def __eq__(self, other):
+        # Just to allow easy testing of equivalent models,
+        # This can be removed once https://github.com/pymc-devs/pytensor/issues/1114 is fixed
+        if type(self) is not type(other):
+            return False
+
+        return equal_computations(
+            self.inner_outputs,
+            other.inner_outputs,
+            self.inner_inputs,
+            other.inner_inputs,
+        )
+
+    def __hash__(self):
```

Comment:
I don't like this. Would it be possible to move this logic into `equal_computations` or a similar place? Also is this actually used anywhere as is?

Reply:
As the comment mentions this will be moved to the base class in PyTensor. It's used for the equal computations check that make one/two tests much more straightforward to write.